### DISCUSSION TITLE
How to adjust line thickness (aka parser seeing empty cells)?

### DISCUSSION THREAD

**Discussion by enrac5** [jsvine/pdfplumber#1092] [Ask for help with specific PDFs]
I'm trying to parse a PDF (see attached) 
[Test_R1.pdf](https://github.com/jsvine/pdfplumber/files/14213383/Test_R1.pdf)

`def process_PDF(pdf_path: str):
    with pdfplumber.open(ccsl_pdf_path) as pdf:
        for page in pdf.pages:
            tables = page.extract_tables()
                for table in tables:
                    for row in table:
                        print(row)
`

I get a printout like this:

['COMBINED CONTINUITY & DIALOGUE', None, None, None, None, 'MASTER ENGLISH SUBTITLE/SPOTTING LIST', None, None, None, None]
['Shot', 'Footage/\nTimecode', None, 'Shot Description/Dialogue', None, 'Title', 'Start', 'End', 'Dur.', 'Subtitle/Spotting']
['', None, None, None, None, None, None, None, None, None]

Notice the second row, there's a 'None' entry after Timecode. The table debugger shows this image (attached).
![tmpatcgtdy7](https://github.com/jsvine/pdfplumber/assets/59802389/cf647ca3-7df1-4e3e-960a-11b75f47cc04)

Which doesn't immediately explain to me why I'm seeing that output, does anyone have any ideas what I'm doing wrong here? For what it's worth, other PDFs (all word docs printed to PDF) don't seem to behave like this.

**Comment by jsvine** [comment]
Hi @enrac5, and thanks for your interest in this library. Semi-regular tables like this can be a bit difficult to work with at first. The key to understanding it is that there is an entry in each row array for each cell horizontal "position" in the table. 

In this case, the right border of the `Footage/Timecode` border is slightly to the left of the that of the timestamped cells at the bottom of the page.

It's not ideal, and something I'd like to make more intuitive / better-structured in future versions of `pdfplumber`.

**Reply by enrac5** [reply]
Hi there, thanks for the response @jsvine, I was able to work around the issue by manipulating the Word doc this PDF came from, so I think I'm okay for now. Had nasty flashbacks to my desktop publishing days (if you remember QuarkXPress, you'll know my pain).
